src/mad_spark_alt/evolution/cost_estimator.py

```
@@ -0,0 +1,658 @@
+
+    def get_adaptive_config(self, base_config: EvolutionConfig) -> EvolutionConfig:
+        """
+        Generate an adapted configuration based on current cost state.
+
+        Args:
+            base_config: Base evolution configuration
+
+        Returns:
+            Adapted configuration optimized for remaining budget
+        """
+        remaining_budget = self.budget_limit - self.accumulated_cost
```

Ruff (0.12.2)

Local variable `remaining_budget` is assigned to but never used

Remove assignment to unused variable `remaining_budget`

(F841)